support-frontend/assets/helpers/productPrice/billingPeriods.ts

```diff
@@ -42,17 +49,92 @@ function billingPeriodTitle(
 		case Quarterly:
 			return fixedTerm ? '3 months' : billingPeriod;
 
+		case OneTime:
+			return 'One-time';
+
 		default:
 			return billingPeriod;
 	}
 }
 
-export {
-	Annual,
-	Monthly,
-	Quarterly,
-	billingPeriodNoun,
-	billingPeriodTitle,
-	weeklyBillingPeriods,
-	weeklyGiftBillingPeriods,
-};
+export function ratePlanToBillingPeriod(
+	ratePlanKey: string | undefined,
+): BillingPeriod {
+	switch (ratePlanKey) {
+		case 'Annual':
+		case 'RestOfWorldAnnual':
+		case 'DomesticAnnual':
+		case 'RestOfWorldAnnualV2':
+		case 'DomesticAnnualV2':
+			return Annual;
+		case 'RestOfWorldQuarterly':
+		case 'DomesticQuarterly':
+			return Quarterly;
+		case 'Monthly':
+		case 'RestOfWorldMonthly':
+		case 'DomesticMonthly':
+		case 'RestOfWorldMonthlyV2':
+		case 'DomesticMonthlyV2':
+		case 'Everyday':
+		case 'Sixday':
+		case 'Weekend':
+		case 'Saturday':
+		case 'Sunday':
+			return Monthly;
+		default:
```

Comment:
Would it be worth being explicit here and matching on the `ratePlayKey` we expect for one time (I guess that's `undefined`?

Reply:
Yes it is, have added `one_off` for readability.

Reply:
ToDo : string replace RatePlanKey

Reply:
`ratePlanKey` has now been replaced with  type `ActiveRatePlanKey` only,  Updated